cmd/neofs-node/cache.go

```diff
@@ -53,47 +53,45 @@ func newNetworkTTLCache(sz int, ttl time.Duration, netRdr netValueReader) *ttlNe
 // updates the value from the network on cache miss or by TTL.
 //
 // returned value should not be modified.
-func (c *ttlNetCache) get(key interface{}) (interface{}, error) {
+func (c *ttlNetCache[K, V]) get(key K) (V, error) {
 	val, ok := c.cache.Peek(key)
 	if ok {
-		valWithTime := val.(*valueWithTime)
-
-		if time.Since(valWithTime.t) < c.ttl {
-			return valWithTime.v, valWithTime.e
+		if time.Since(val.t) < c.ttl {
+			return val.v, val.e
 		}
 
 		c.cache.Remove(key)
 	}
 
-	val, err := c.netRdr(key)
+	netVal, err := c.netRdr(key)
 
-	c.set(key, val, err)
+	c.set(key, netVal, err)
 
-	return val, err
+	return netVal, err
 }
 
-func (c *ttlNetCache) set(k, v interface{}, e error) {
-	c.cache.Add(k, &valueWithTime{
+func (c *ttlNetCache[K, V]) set(k K, v V, e error) {
+	c.cache.Add(k, &valueWithTime[V]{
 		v: v,
 		t: time.Now(),
 		e: e,
 	})
 }
 
-func (c *ttlNetCache) remove(key interface{}) {
+func (c *ttlNetCache[K, V]) remove(key K) {
 	c.cache.Remove(key)
 }
 
 // entity that provides LRU cache interface.
 type lruNetCache struct {
-	cache *lru.Cache
+	cache *lru.Cache[uint64, *netmapSDK.NetMap]
```

Comment:
As much as I'm skeptical about generics, this probably makes things more readable.